### PR TITLE
Fix #14157

### LINKED ISSUE #14157: Add a new geometry node to extractUnique positions from a geometry

### PR DESCRIPTION
**Pull request by deltakosh**
Also making sure we do not rebuild if not required

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14158/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14158/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14158/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14158/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14158/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14158/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14158/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14158/merge#BCU1XR#0

**Review by RaananW — APPROVED** [review]

File: packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                const y = vertexData.positions[index + 1];
+                const z = vertexData.positions[index + 2];
+
+                // check if we already have it
+                let found = false;
+                for (let checkIndex = 0; checkIndex < newPositions.length; checkIndex += 3) {
+                    if (
+                        Scalar.WithinEpsilon(x, newPositions[checkIndex], Epsilon) &&
+                        Scalar.WithinEpsilon(y, newPositions[checkIndex + 1], Epsilon) &&
+                        Scalar.WithinEpsilon(z, newPositions[checkIndex + 2], Epsilon)
```
Comment (RaananW) [inline packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts:75]: Just a thought (can be resolved) - would it be possible to configure the epsilon (defaults to the static Epsilon value) to allow the user a more precise "unique" definition?

Comment (deltakosh) [reply packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts:75]: What do you mean? (Sorry I'm slow)

Comment (deltakosh) [reply packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts:75]: oh you mean on the block itself? oh that's a good idea!

Comment (RaananW) [reply packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts:75]: You are testing positions against already-existing positions in the array, based on an epsilon value. This epsilon is fixed, so theoretically, if the mesh's positions are very small a lot of them will be included and the mesh will be "decimated". If we can configure the epsilon we could define the precision of the compare function

**Pull request merged**